cli/tests/pcluster/templates/test_cluster_stack.py

```diff
@@ -43,3 +50,47 @@ def test_awsbatch_cluster_builder(mocker):
     )
     print(yaml.dump(generated_template))
     # TODO assert content of the template by matching expected template
+
+
+@pytest.mark.parametrize(
+    "config_file_name, expected_head_node_dna_json_file_name",
+    [
+        ("slurm-imds-secured-true.yaml", "slurm-imds-secured-true.head-node.dna.json"),
+        ("slurm-imds-secured-false.yaml", "slurm-imds-secured-false.head-node.dna.json"),
+        ("awsbatch-imds-secured-false.yaml", "awsbatch-imds-secured-false.head-node.dna.json"),
+    ],
+)
+@freeze_time("2021-01-01T01:01:01")
+def test_head_node_dna_json(mocker, test_datadir, config_file_name, expected_head_node_dna_json_file_name):
+    mock_aws_api(mocker)
+
+    input_yaml = load_yaml_dict(test_datadir / config_file_name)
+
+    cluster_config = ClusterSchema().load(input_yaml)
+
+    generated_template = CDKTemplateBuilder().build_cluster_template(
+        cluster_config=cluster_config,
+        bucket=dummy_cluster_bucket(),
+        stack_name="parallelcluster-dummyname",
+    )
+
+    generated_head_node_dna_json = json.loads(
+        _get_cfn_init_file_content(template=generated_template, resource="HeadNodeLaunchTemplate", file="/tmp/dna.json")
+    )
+    expected_head_node_dna_json = load_json_dict(test_datadir / expected_head_node_dna_json_file_name)
+
+    diff = DeepDiff(
+        expected_head_node_dna_json,
+        generated_head_node_dna_json,
+        ignore_order=True,
+    )
+
+    assert not diff
```

Comment:
* can [assertpy](https://github.com/assertpy/assertpy) be used instead of the builtin assertion? it's used throughout the rest of the tests, and helps keep a constant experience for the stack traces resulting in test failures.
* rather than adding a new dependency on `deepdiff`, can `assertpy`'s [dict comparisons](https://github.com/assertpy/assertpy#dicts) be used?

Reply:
Sure, good catch!